conn.go

```diff
@@ -264,6 +264,15 @@ func (l *Conn) StartTLS(config *tls.Config) error {
 	return nil
 }
 
+// TLSConnectionState returns the current connection's TLS ConnectionState, or an error if not TLS.
+func (l *Conn) TLSConnectionState() (*tls.ConnectionState, error) {
```

Comment:
Exposing the connection seems like the wrong thing to me:  The actual I/O to the server should always be handled by the protocol client, so you don't really want to expose the connection itself.

I think we're going to continue using our vendored fork instead, so I'm going to go ahead and close this PR, as I don't have time allocated to do further dev here.